test/units/modules/network/avi/test_avi_user.py

```diff
@@ -0,0 +1,107 @@
+import os
+import json
+from units.compat import unittest
+from units.compat.mock import Mock
+from units.modules.utils import set_module_args
+from ansible.modules.network.avi import avi_user as avi_user
+
+fixture_path = os.path.join(os.path.dirname(__file__), 'fixtures')
+with open(fixture_path +'/avi_user.json') as json_file:
+    data = json.load(json_file)
+
+class TestAviUser(unittest.TestCase):
+
+    def test_create_user(self):
+        set_module_args({
+                "avi_credentials": {
+                    "controller": "10.23.45.13",
+                    "username": "username",
+                    "password": "fakepassword",
+                    "api_version": "18.2.5"
+
+                },
+                "state": "present",
+                "name": "testuser",
+                "obj_username": "testuser",
+                "obj_password": "test123",
+                "email": "test@abc.com",
+                "access":[
+                    {
+                        "role_ref": "/api/role?name=Tenant-Admin",
+                        "tenant_ref": "/api/tenant?name=Test-Admin",
+                        "all_tenants": False
+                    }
+                ],
+                "user_profile_ref": "/api/useraccountprofile?name=Default-User-Account-Profile",
+                "is_active": True,
+                "is_superuser": True,
+                "default_tenant_ref": "/api/tenant?name=admin"
+            })
+
+        avi_user.avi_ansible_api = Mock(return_value=data['mock_create_res'])
+        responce = avi_user.avi_ansible_api()
```

Comment:
I thought we should be testing the module invocation and not directly testing the return of the mocked function avi_ansible_api